### PR TITLE
Added gen_help_text.go and docs/ directory for cobra generated help d…

### PR DESCRIPTION
**Pull request by aaron-prindle**
…ocumentation

Please comment on how to make the document generation be automatic for each pull.  Right now this is run outside of the context of minikube via 'go run gen_help_text.go'.  Also any suggestion on how to test this would be appreciated, thanks.

#171

### CODE REVIEW

**Comment by dlorenc** [comment]
Nice! One idea would be to add a step to test.sh that runs "go run gen_help_text.go" and checks to see if any files have been changed.

**Comment by dlorenc** [comment]
Something like this would probably work:

```shell
if [[ $(git diff) ]]; then
  echo "Help text is out of date. Please run "go run gen_help_text.go"
  exit 1
else
  echo "Help text is up to date"
fi
```

File: test.sh [hunk 1]
```diff
@@ -49,3 +49,13 @@ if [ ! -z ${files} ]; then
 	echo "Boilerplate missing in: ${files}."
 	exit 1
 fi
+
+# Check that cobra docs are up to date
+# This is done by generating new docs and then seeing if they are different than the committed docs
+go run gen_help_text.go
+if [[ $(git diff) ]]; then
```
Comment (dlorenc) [inline test.sh]: Just thought of this, but could you store this result in a variable? That way you can echo it out in the error message as well.

```shell
files=$(git diff)
if [[ $files ]]; then
  echo "Help text is out of date... $files"
```

**Comment by dlorenc** [comment]
Awesome! Could you add a link to the top-level docs page from the main README.md?

**Comment by aaron-prindle** [comment]
I added some text to the README.md with a section titled Documentation and a link to the main minikube.md file which links to all of its subcommands, please let me know if I should reword or restructure it, thanks.

**Comment by dlorenc** [comment]
LGTM. I'll merge once travis is happy.

**Pull request merged**